base_layer/core/tests/horizon_state_validators.rs

```diff
@@ -141,3 +168,181 @@ fn validate_achieved_difficulty() {
     assert!(store.commit(txn).is_ok());
     assert!(store.validate_horizon_state().is_err());
 }
+
+#[test]
+fn validate_chain_genesis_block() {
+    let validators = Validators::new(
+        MockValidator::new(true),
+        MockValidator::new(true),
+        MockValidator::new(true),
+        ChainGbValidator::new(),
+        MockValidator::new(true),
+    );
+    let factories = Arc::new(CryptoFactories::default());
+
+    let db = MemoryDatabase::<HashDigest>::default();
+    let mut store = BlockchainDatabase::new(db).unwrap();
+    store.set_validators(validators.clone());
+    let (block0, _) = create_genesis_block(&store, &factories);
+    store.add_block(block0).unwrap();
+    assert_eq!(
+        store.validate_horizon_state(),
+        Err(ChainStorageError::ValidationError(ValidationError::BlockHeaderError(
+            BlockHeaderValidationError::IncorrectGenesisBlockHeader
+        )))
+    );
+
+    let db = MemoryDatabase::<HashDigest>::default();
+    let mut store = BlockchainDatabase::new(db).unwrap();
+    store.set_validators(validators);
+    store.add_block(get_genesis_block()).unwrap();
+    assert!(store.validate_horizon_state().is_ok());
+}
+
+fn create_test_blockchain_with_emmision(
+    rules: &ConsensusManager<MemoryDatabase<HashDigest>>,
+    factories: &CryptoFactories,
+    store: &mut BlockchainDatabase<MemoryDatabase<HashDigest>>,
+) -> (Vec<Block>, Vec<Vec<UnblindedOutput>>)
+{
+    // Block 0
+    let block_reward0 = rules.emission_schedule().block_reward(0);
+    let (block0, output) = create_genesis_block_with_coinbase_value(&store, &factories, block_reward0);
+    store.add_block(block0.clone()).unwrap();
+    let mut blocks = vec![block0];
+    let mut outputs = vec![vec![output]];
+
+    // Block 1
+    let fee_per_gram = 25 * uT;
+    let total_fee = Fee::calculate(fee_per_gram, 1, 2);
+    let transfer_amount1 = 1000 * uT;
+    let transfer_amount2 = block_reward0 - transfer_amount1 - total_fee;
+    let schema = vec![txn_schema!(
+        from: vec![outputs[0][0].clone()],
+        to: vec![transfer_amount1, transfer_amount2],
```

Comment:
FYI - the txn_schema macro _should_ do this for you if you leave it out. i.e. a change output is automatically created. If not, it's a bug.

Reply:
That is great, I changed the test to make use of the change output provided by the txn_schema macro.